table/block_based/block.cc

```diff
@@ -659,17 +648,70 @@ void IndexBlockIter::DecodeCurrentValue(uint32_t shared) {
   }
 }
 
-// Binary search in restart array to find the first restart point that
-// is either the last restart point with a key less than target,
-// which means the key of next restart point is larger than target, or
-// the first restart point with a key = target
+template <class TValue>
+void BlockIter<TValue>::ScanAfterBinarySeek(const Slice& target, uint32_t index,
+                                            bool is_index_key_result,
+                                            const Comparator* comp) {
+  // Linear search (within restart block) for first key >= target
+  if (is_index_key_result) {
+    SeekToRestartPoint(index);
+    Next();
+  } else {
+    SeekToRestartPoint(index);
+    uint32_t max_offset;
+    if (index + 1 < num_restarts_) {
+      max_offset = GetRestartPoint(index + 1);
+    } else {
+      max_offset = restarts_;
```

Comment:
I don't think the `current_ == max_offset` case below is hit under this assignment, so in theory it could be `max_offset = restarts_ + 12345;`

I don't have a suggested code improvement, but maybe worth a code comment about intentions

Reply:
Done. Made it `kMaxUint32` so it's extra clear it's not intended to be the loop terminating condition.